pkg/build/builder/common.go

```diff
@@ -180,3 +155,47 @@ func execPostCommitHook(client DockerClient, postCommitSpec api.BuildPostCommitS
 		Stderr:       true,
 	})
 }
+
+func updateBuildRevision(build *api.Build, sourceInfo *git.SourceInfo) *api.SourceRevision {
+	if build.Spec.Revision != nil {
+		return build.Spec.Revision
+	}
+	return &api.SourceRevision{
+		Git: &api.GitSourceRevision{
+			Commit:  sourceInfo.CommitID,
+			Message: sourceInfo.Message,
+			Author: api.SourceControlUser{
+				Name:  sourceInfo.AuthorName,
+				Email: sourceInfo.AuthorEmail,
+			},
+			Committer: api.SourceControlUser{
+				Name:  sourceInfo.CommitterName,
+				Email: sourceInfo.CommitterEmail,
+			},
+		},
+	}
+}
+
+func updateStatusRetrier(build *api.Build, client client.BuildInterface, sourceRev *api.SourceRevision) error {
+	return kclient.RetryOnConflict(kclient.DefaultBackoff, func() error {
```

Comment:
yes technically it's the builder image, but realistically we can consider actions the builder image takes to be direct extensions of actions the controller takes.  And it is the primary object (the build object, owned by the controller) that's being updated.

So it seems like using update would follow the existing patterns, such as they are.